docker/docker_image_src.go

```diff
@@ -28,17 +29,70 @@ type dockerImageSource struct {
 	cachedManifestMIMEType string // Only valid if cachedManifest != nil
 }
 
-// newImageSource creates a new ImageSource for the specified image reference.
-// The caller must call .Close() on the returned ImageSource.
-func newImageSource(sys *types.SystemContext, ref dockerReference) (*dockerImageSource, error) {
-	c, err := newDockerClientFromRef(sys, ref, false, "pull")
+// newImageSource creates a new `ImageSource` for the specified image reference
+// `ref`.
+//
+// During the creation of the instance, all provided registry mirrors within
+// `/etc/containers/registries.conf` (the default location) will be tested for
+// the availability of the image manifest. References which contain a
+// configured prefix will be automatically rewritten to the correct target
+// reference.
+//
+// If no mirror is available in the config, then the given `ref` will be used
+// to verify that the image is available at the remote location.
```

Comment:
This reads like that if we have no mirrors, we are checking for the existence of `ref` which is not the case (due to the prefix semantics). I think we can rephrase the comments by splitting the mechanism into:
1. Doing a lookup for configured registries in the `registries.conf`
2. If a prefix is specified, the reference will be rewritten to match the specified endpoint (give an example).
3. If the rewritten reference exists, we use it as the `dockerRef` in the client. Otherwise, we fall back to using the initial `ref`
4. Mention that mirrors are tested (in the specified order) prior to the endpoint and the fallback